train.py

```diff
@@ -426,15 +429,23 @@ def __generate_test_samples(pipe, prompts, gs, log_writer, log_folder, random_ca
             del tfimage
             del images
 
-    try: 
-        hf_ckpt_path = convert_to_hf(args.resume_ckpt)
-        text_encoder = CLIPTextModel.from_pretrained(hf_ckpt_path, subfolder="text_encoder")
-        vae = AutoencoderKL.from_pretrained(hf_ckpt_path, subfolder="vae")
-        unet = UNet2DConditionModel.from_pretrained(hf_ckpt_path, subfolder="unet")
-        sample_scheduler = DDIMScheduler.from_pretrained(hf_ckpt_path, subfolder="scheduler")
-        noise_scheduler = DDPMScheduler.from_pretrained(hf_ckpt_path, subfolder="scheduler")
-        tokenizer = CLIPTokenizer.from_pretrained(hf_ckpt_path, subfolder="tokenizer", use_fast=False)
-    except:
+    try:
+        # first try to download from HF
+        model_root_folder = try_download_model_from_hf(repo_id=args.resume_ckpt,
```

Comment:
What happens here when `args.hf_access_token is None`?

Reply:
if the repo doesn't need a token, it just downloads. if the repo needs a token and one is already set in the user's environment (huggingface_hub has a number of ways it does this i believe), then it just downloads. otherwise the huggingface lib raises and exception and bails out.